### ISSUE TITLE
Regression in 1.9.x series for --variable / --define-variable not overriding variables

### ISSUE THREAD

**Issue by eli-schwartz** [pkgconf/pkgconf#259]
Using the following git bisect test line

```
meson setup builddir -Dtests=false --prefix=/usr --wipe && \
    ninja -C builddir && \
    ./builddir/pkgconf zlib --define-variable=prefix=/tmp --variable=prefix
```

I expect to have the value of "prefix" overridden to /tmp, and then printed out.

Starting from commit d68a867dc0297b51925d7d4ecc6704b147fd7116 this is no longer true -- it prints "/usr" instead.

Trying to define prefix to /tmp, and print the variable "libdir", still works correctly -- I get /tmp/lib in that case.

This breaks Meson CI, as we have a test to see if `pkgconfig_define:` works by checking a known-installed dependency (zlib) and defining the variable we print (because we cannot necessarily predict e.g. libdir is `${prefix}/lib` instead of lib64 or something).

**Comment by kaniini** [comment]
ah, it makes sense, the `prefix` variable is overridden.  i will add some code to mark global vars as overridden.

**Comment by kaniini** [comment]
this should now be resolved in the pending 1.9.3 release
